lib/SILOptimizer/Utils/OwnershipOptUtils.cpp

```diff
@@ -56,7 +57,34 @@ replaceAllUsesAndEraseInner(SingleValueInstruction *svi, SILValue newValue,
         user->eraseFromParent();
       continue;
     }
+
+    // Special handling for copy_value with opened archetypes
+    // When replacing a copy_value's operand with a newValue, if the newValue's
+    // opened archetype is different from copy_value's opened archetype, we have
+    // to clone the copy_value, so that we have the correct opened archetype on
+    // copy_value's result type.
+    bool needsCloning = false;
```

Comment:
What is causing this to be needed? The reason I am asking is rather than expand what this routine does, I wonder if we could instead put the burden in some way on a different part of the code. Just the context would be useful.

Reply:
I removed handling open_existential_ref in CSE. I'll follow this up with another PR with spl handling needed to deal with type dependent operands.